### ISSUE TITLE
Devices that do not include a service UUID in advertising data cannot be detected on macOS 12.0 to 12.2 unless running in app created by py2app

### ISSUE THREAD

**Issue by dlech** [hbldh/bleak#720]
* bleak version: 0.14.0
* Python version: all
* Operating System: macOS 12.x

### Description

This is basically the same issue as #635. That issue was closed because there is now a workaround available in Bleak v0.14.0 that works in most cases. However, there is one case that doesn't have an acceptable workaround. If a device does not include a service UUID in the advertising data, then the advertising data from that device cannot be received in Bleak.

### Call to action

This is a regression from previous versions of macOS. Please everyone report this issue to Apple using the Feedback Assistant app. If enough people report it, hopefully they will prioritize fixing it.
### Workaround

This isn't great, but it is possible to convert your Python script to an app using [py2app](https://py2app.readthedocs.io/en/latest/index.html) to work around the issue. Here is an example `setup.py` that can be uses as a starting point:

```python
from setuptools import setup

setup(
    app=["my_script.py"],
    setup_requires=["py2app"],
    options=dict(
        py2app=dict(
            plist=dict(
                NSBluetoothAlwaysUsageDescription="This app uses Bluetooth.",
            ),
        ),
    ),
)
```

Then run `python setup.py py2app` to build the .app.

If your script does not have a graphical user interface, you can run it with:

    open -n -W --stdin $TTY --stdout $TTY --stderr $TTY dist/my_script.app

See `man open` for more info.

Control C doesn't work to stop the program, but you can secondary-click on the icon that is created in the dock and force quit if needed.

**Comment by hbldh** [comment]
Very good to keep an issue alive for this.

**Comment by pauloesteban** [comment]
Hi everyone,

Anyone tried a similar workaround with PyInstaller?

Cheers

**Comment by kdewald** [comment]
@dlech have you have any updates on your research on this topic?

I've been able to reproduce it with a very minimalist console application on XCode (https://github.com/kdewald/MontereyBluetoothError). My theory is that there is some problem between the TCC and the Bluetooth subsystems on Monterey, as looking at the logs shows that certain authorization steps do not happen when running console applications, even though they have Bluetooth authorization.

**Comment by dlech** [comment]
I haven't done any further research. Reiterating what I have already said, the `NSBluetoothAlwaysUsageDescription` (I assume this is what you mean when you say "they have Bluetooth authorization") only prevents the app from crashing with `SIGABRT`. It is completely independent of the "no scan results" problem. The "no scan results" problem has only to do with the ability of the process to be considered the "foreground" app which consists of having an icon in the dock and being properly signed.

**Comment by dlech** [comment]
Good news! It looks like Apple has fixed the issue in macOS 12.3 Beta (21E5196i).
